Autopilot/DataStructures/Src/vectorClass.cpp

```diff
@@ -0,0 +1,108 @@
+/* 
+    Author: Nixon Chan
+*/
+
+#include <math.h>
+#include "vectorClass.hpp"
+
+Vector3D::Vector3D()
+{
+    x = 0;
+    y = 0;
+    z = 0;
+}
+
+Vector3D::Vector3D(double inputX, double inputY, double inputZ)
+{
+    x = inputX;
+    y = inputY;
+    z = inputZ;
+}
+
+Vector3D::~Vector3D()
+{
+}
+
+double Vector3D::getX()
+{
+    return x;
+}
+
+double Vector3D::getY() 
+{
+    return y;
+}
+
+double Vector3D::getZ()
+{
+    return z;
+}
+
+void Vector3D::setX(double input)
+{
+    x = input;
+}
+
+void Vector3D::setY(double input)
+{
+    y = input;
+}
+
+void Vector3D::setZ(double input)
+{
+    z = input;
+}
+
+Vector3D& Vector3D::operator=(const Vector3D &vector2)
+{
+    (*this).x = vector2.x;
+    (*this).y = vector2.y;
+    (*this).z = vector2.z;
+    return *this;
```

Comment:
This function returns the address to the Vector3D object, yet we are returning `*this`. I think my knowledge of pointers is breaking down cause I'm pretty confused. Mind explaining what's going on here?

Reply:
*this refers to the object which the function is called on. I use another method if its slightly confusing.

Reply:
^wait scratch what I said, it works with and without the reference somehow. I changed it so that it returns the original object, so that we can use it for operator chaining.